src/index.js

```diff
@@ -0,0 +1,154 @@
+    const octokit = getOctokit(token);
+
+    const {
+        owner,
+        repo,
+    } = context.repo;
+
+    const milestones = await octokit.paginate(octokit.rest.issues.listMilestones, {
+        owner,
+        repo,
+    });
+
+    const milestoneInfo = milestones.find(({title}) => {
+        return title === milestone;
+    });
+
+    switch (context.payload.action) {
+        case events.CREATE_MILESTONE: {
+            if (milestoneInfo) {
+                throw new Error(`Can't create milestone, ${milestone} version already exists`);
+            }
+
+            const due_on = getDueOn(core.getInput('due_on', {required: false}));
+            const description = core.getInput('description', {required: false});
+
+            await octokit.rest.issues.createMilestone({
+                owner,
+                repo,
+                title: milestone,
+                description,
+                due_on,
+            });
+
+            break;
+        }
+
+        case events.CLOSE_MILESTONE: {
+            if (!milestoneInfo) {
+                throw new Error(`Can't close milestone, ${milestone} version does not exists`);
+            }
+
+            await octokit.rest.issues.updateMilestone({
+                owner,
+                repo,
+                milestone_number: milestoneInfo.number,
+                state: 'closed',
+            });
+
+            break;
+        }
+
+        case events.UPDATE_MILESTONE: {
+            if (!milestoneInfo) {
+                throw new Error(`Can't update milestone, ${milestone} version does not exists`);
+            }
+
+            const due_on = getDueOn(core.getInput('due_on', {required: false}));
+            const description = core.getInput('description', {required: false});
+
+            if(
+                (description === milestoneInfo.description) ||
+                (due_on === milestoneInfo.due_on)
+            ) {
+                core.info('Description or due on fields didn\'t changed, do nothing');
+                process.exit(0);
+            }
+
+            await octokit.rest.issues.updateMilestone({
+                owner,
+                repo,
+                milestone_number: milestoneInfo.number,
+                description,
+                due_on,
+            });
+
+            break;
+        }
+
+        case events.ASSIGN_MILESTONE: {
+            if (!milestoneInfo) {
+                throw new Error(`Can't assign milestone to pull-request, ${milestone} version does not exists`);
+            }
+
+            const issue = core.getInput('issue', {required: false});
+
+            if (!issue) {
+                throw new Error(`Can't assign milestone to pull-request, issue parameter is missing`);
+            }
+
+            // see https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-issues-and-pull-requests
+            const q = `${issue} is:pr in:title repo:${owner}/${repo}`;
```

Comment:
I think the naming `q` is bad. maybe `query`, if thats what you mean?

Reply:
I tend to agree with you but the api [api](https://docs.github.com/en/rest/reference/search#search-issues-and-pull-requests) require `q` as parameter, so I tried to safe some code